shell/shared/grain-shared.js

```diff
@@ -57,6 +57,21 @@ Meteor.methods({
                         trashed: { $exists: false }, },
                        { $set: { "trashed": new Date() } },
                        { multi: true });
+
+      const grainsOwned = Grains.find({
+        userId: { $eq: this.userId },
+        _id: { $in: grainIds },
+      }, { fields: { _id: 1, }, });
+      grainsOwned.forEach((grain) => {
+        try {
+          Sessions.remove({ grainId: grain._id, });
```

Comment:
Theoretically mongo operations could throw exceptions, although I will say we rarely ever catch/deal with them...